### DISCUSSION TITLE
2 NIC ltsp cluster

### DISCUSSION THREAD

**Discussion by stge4code** [ltsp/ltsp#331] [Q&A]
Dear developers, thank you for this useful tool!

May I ask you to help me with setting ltsp cluster with two separate networks.  
I have Ubuntu 20.04.1 server with two different NIC's: one (1stNIC) connected to Internet thought router with its own DHCP (192.168.1.x)  the other (2nd NIC) connected though switch to local clients.  I tied to manually set IP address 192.168.67.1 to 2ndNIC and run **sudo ltsp dnsmasq --proxy-dhcp=0**,  **sudo ltsp dnsmasq --real-dhcp=1** and **sudo ltsp dnsmasq**  without any success. After these commands ltsp says about copying dnsmasq conf files and restarting dnsmasq service but DHCP does not start within internal network and 2ndNIC IP (192.168.67.1)  remains unchanged. It seems to me ltsp is not able to detect 2ndNIC and try to use 1stNIC. Is it any way to directly tell ltsp which NIC it should be use? 

When I do the same steps on my virtual machine (Virtual Box 6, Ubuntu 20.04.1 guest, Windoows 10 host) everything works fine.

Kind regars,
Alexey

**Comment by alkisg** [comment]
Run `sudo ltsp dnsmasq --proxy-dhcp=0` and leave it like that. Then run `sudo netstat -nap | grep dnsmasq` and paste the output here, it should list dnsmasq listening on 0.0.0.0:67, 69 and 4011. This means that dnsmasq actually listens on ALL interfaces anyway, but it just answers in 192.168.67.1.

Then boot a client inside the LTSP network, even with a live USB image. Does it get an IP? If not, maybe you're running some firewall...

**Reply by stge4code** [reply]
Thank you for a quick reply!
I've done this and get the following:
```
$ sudo ltsp dnsmasq --proxy-dhcp=0
Installed /usr/share/ltsp/server/dnsmasq/ltsp-dnsmasq.conf in /etc/dnsmasq.d/ltsp-dnsmasq.conf
Restarted dnsmasq
$ sudo netstat -nap | grep dnsmasq
udp        0      0 0.0.0.0:67              0.0.0.0:*                           2689/dnsmasq        
udp        0      0 0.0.0.0:69              0.0.0.0:*                           2689/dnsmasq        
udp        0      0 0.0.0.0:4011            0.0.0.0:*                           2689/dnsmasq        
udp6       0      0 :::69                   :::*                                2689/dnsmasq        
unix  2      [ ]         DGRAM                    65118    2689/dnsmasq     
```

Also when 1stNIC physically disconnected with router (and Internet) I got this:

```
$ sudo ltsp dnsmasq --proxy-dhcp=0
RTNETLINK answers: Network is unreachable
LTSP command failed: ip -o route get 192.168.67.2
Could not parse output of: ip -o route get 192.168.67.2
Installed /usr/share/ltsp/server/dnsmasq/ltsp-dnsmasq.conf in /etc/dnsmasq.d/ltsp-dnsmasq.conf
Restarted dnsmasq
```

My Ubuntu20.04.1 is up-to-date and  installed without any additional packages (only mc and grub-customizer).
Firewall **ufw** is inactive.

I try to load clients with Ubuntu Live USB, but they cannot obtain IP's. The internal network (server 2ndNIC->1Gb switch->client) is unreachable.

**Comment by alkisg** [comment]
I think you didn't manage to set a static IP.

The following command shows your IP. If you do set a static IP in eth2, it should say:

```shell
$ ip a
3: eth2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether c0:4a:00:02:bc:1e brd ff:ff:ff:ff:ff:ff
    inet 192.168.67.1/24 brd 192.168.67.255 scope global noprefixroute eth2
```

The following command shows the route to 192.168.67.2. If you do set a static IP in eth2, it should say:

```shell
$ ip - o route get 192.168.67.2
192.168.67.2 dev eth2 src 192.168.67.1 uid 1010 \    cache 
```

What do these commands say for you?

**Reply by stge4code** [reply]
I've set static IP by using Network-Manager GUI  (upper right icon on Ubuntu Desktop panel). Also, as I understand Ubuntu uses netplan and different NIC names like enpXsY.

```
$ ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: enp5s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether bc:ae:c7:29:c3:96 brd ff:ff:ff:ff:ff:ff
    inet 192.168.67.1/0 brd 255.255.255.255 scope global noprefixroute enp5s0
       valid_lft forever preferred_lft forever
    inet6 fe80::56e2:92d5:2eb3:3928/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever
3: enp7s1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 00:18:9a:38:21:9b brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.34/24 brd 192.168.1.255 scope global dynamic noprefixroute enp7s1
       valid_lft 21924sec preferred_lft 21924sec
    inet6 fe80::63ge:64a3:1579:1703/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever
```

```
$ sudo ufw status
[sudo] password for user: 
Status: inactive
$ ip - o route get 192.168.67.2
192.168.67.2 via 192.168.1.1 dev enp7s1 src 192.168.1.34 uid 1000 
    cache 
$ cat /var/lib/misc/dnsmasq.leases
$ 
```

**Comment by alkisg** [comment]
You set the wrong subnet mask for 192.168.67.1, it should be 255.255.255.0.

You can use triple backticks, ```, for terminal output, I edited your message, you can see how I did it.

**Reply by stge4code** [reply]
Thank you **alkisg**! 
I've changed the mask, reboot switch which is connected to the 2ndNIC. It seems that server DHCP start working now.

P.S. How do you think is ltsp appropriate for building openmpi-based computational cluster like unsupported PelicanHPC or Kestrel? Is it any way to install additional packages directly to clients machine without rebuilding ltsp image.
Best regards

**Comment by alkisg** [comment]
I think LTSP is appropriate for such clusters, and I've heard of similar successful cases in the past.
To add programs to the clients you always need to rebuild the LTSP image. This is actually the greatest LTSP benefit because you only do it once instead of installing the program in each client separately.
